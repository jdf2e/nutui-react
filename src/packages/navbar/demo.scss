.demo {
  .flex-center {
    display: inline-flex;
    align-items: center;
  }

  .nut-navbar-title .title {
    display: flex;
    flex-direction: column;

    .desc {
      font-weight: 400;
      font-size: 12px;
      color: $color-text-help;
    }
  }

  .navbar-tabs {
    .nut-tabs-titles {
      padding: 0;
      .nut-tabs-titles-item {
        margin: 0;
        font-size: 16px;

        &-active {
          font-size: 20px;
        }
      }
    }
  }
}
