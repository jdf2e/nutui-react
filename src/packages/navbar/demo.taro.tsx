import React from 'react'
import { useTranslate } from '@/sites/assets/locale/taro'
import { NavBar } from '@/packages/nutui.react.taro'

interface T {
  ce5c5446: string
  c38a08ef: string
  b840c88f: string
  a74a1fd4: string
  '8dab2f66': string
  cfbdc781: string
  c3a3a1d2: string
  e51e4582: string
}

const NavBarDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      ce5c5446: '基础用法',
      c38a08ef: '订单详情',
      b840c88f: '标题',
      a74a1fd4: '返回',
      '8dab2f66': '编辑',
      cfbdc781: '清空',
      c3a3a1d2: '购物车',
      e51e4582: '浏览记录',
    },
    'zh-TW': {
      ce5c5446: '基礎用法',
      c38a08ef: '訂單詳情',
      b840c88f: '標題',
      a74a1fd4: '返回',
      '8dab2f66': '編輯',
      cfbdc781: '清空',
      c3a3a1d2: '購物車',
      e51e4582: '瀏覽記錄',
    },
    'en-US': {
      ce5c5446: 'Basic usage',
      c38a08ef: 'order details',
      b840c88f: 'title',
      a74a1fd4: 'return',
      '8dab2f66': 'edit',
      cfbdc781: 'empty',
      c3a3a1d2: 'shopping cart',
      e51e4582: 'Browsing history',
    },
  })
  return (
    <>
      <div className="demo">
        <h2>{translated.ce5c5446}</h2>
        <NavBar
          title={translated.c38a08ef}
          icon="share"
          leftShow
          onClickTitle={(e) => alert(translated.b840c88f)}
          onClickBack={(e) => alert(translated.a74a1fd4)}
          onClickIcon={(e) => alert('icon')}
        />
        <NavBar
          title={translated.e51e4582}
          desc={translated.cfbdc781}
          leftShow
          onClickTitle={(e) => alert(translated.b840c88f)}
          onClickBack={(e) => alert(translated.a74a1fd4)}
          onClickClear={(e) => alert(translated.cfbdc781)}
        />
        <NavBar
          title={translated.c3a3a1d2}
          icon="more"
          desc={translated['8dab2f66']}
          titIcon="locationg3"
          onClickTitle={(e) => alert(translated.b840c88f)}
          onClickBack={(e) => alert(translated.a74a1fd4)}
          onClickClear={(e) => alert(translated['8dab2f66'])}
          onClickIcon={(e) => alert('icon')}
        />
      </div>
    </>
  )
}

export default NavBarDemo
