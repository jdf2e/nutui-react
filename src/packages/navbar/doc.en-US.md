# Navbar

### introduce 


Provides navigation capabilities.

### Install

```ts
import { NavBar } from '@nutui/nutui-react';
```

### code example

### Basic usage

:::demo
```tsx
import React from "react";
import { NavBar, Icon } from '@nutui/nutui-react';

const App = () => {
  return ( 
    <>   
      <NavBar
          title="order details"
          leftShow
          leftText="back"
          onClickTitle={(e) => alert("back")}
          onClickBack={(e) => alert("title")}
          onClickRight={(e) => alert('icon')}
        >
          <Icon name="share" slot="right" />
        </NavBar>
    </>
  );
};  
export default App;

```
:::

:::demo
```tsx
import  React from "react";
import { NavBar } from '@nutui/nutui-react';

const App = () => {
  return ( 
    <>   
      <NavBar
          title="Browsing history"
          desc="clear"
          leftShow
          onClickTitle={(e) => alert("back")}
          onClickBack={(e) => alert("title")}
          onClickRight={(e) => alert('clear')}
      />
    </>
  );
};  
export default App;

```
:::

:::demo
```tsx
import  React from "react";
import { NavBar, Icon } from '@nutui/nutui-react';

const App = () => {
  return ( 
    <>   
      <NavBar
          title="cart"
          desc="edit"
          titIcon="locationg3"
          onClickTitle={(e) => alert("back")}
          onClickBack={(e) => alert("title")}
          onClickRight={(e) => alert('edit')}
          onClickIcon={(e) => alert('icon')}
        >
          <Icon name="more-x" slot="right" />
      </NavBar>
    </>
  );
};  
export default App;

```
:::

:::demo
```tsx
import  React from "react";
import { NavBar, Icon } from '@nutui/nutui-react';

const App = () => {
  return ( 
    <>
      <NavBar
          title="order details"
          leftShow
          border
          leftText="back"
          onClickTitle={(e) => alert("back")}
          onClickBack={(e) => alert("title")}
          onClickRight={(e) => alert('icon')}
        >
          <Icon name="share" slot="right" />
      </NavBar>
    </>
  );
};  
export default App;

```
:::


### Customize the middle content of the navigation bar

:::demo
```tsx
import  React, { useState } from "react";
import { NavBar, Icon, Tabs, TabPane } from '@nutui/nutui-react';

const App = () => {
  const [tab1value, setTab1value] = useState('Tab 1')
  return ( 
    <>   
      <NavBar
          desc="edit"
          onClickTitle={(e) => alert("title")}
          onClickRight={(e) => alert("edit")}
          onClickBack={(e) => alert("back")}
          onClickIcon={(e) => alert('icon')}
        >
          <div slot="content">
            <Tabs value={tab1value} onChange={({ paneKey }) => { setTab1value(paneKey) }}>
              <TabPane title="Tab 1"> Tab 1 </TabPane>
              <TabPane title="Tab 2"> Tab 2 </TabPane>
              <TabPane title="Tab 3"> Tab 3 </TabPane>
            </Tabs>
          </div>
          <Icon name="more-x" slot="right" />
      </NavBar>
    </>
  );
};  
export default App;

```
:::

### Prop  

| Prop            | Description                                                                                           | Type    | Default  |
|-----------------|------------------------------------------------------------------------------------------------|---------|---------|
| title           | title name                                                                                       | String  | -       |
| desc            | Description on the right                                                                                       | String  | -       |
| leftShow        | Whether to show the left arrow                                                                              | Boolean | true   |
| titIcon         | title with icon                                                         | String  | -       |   
| leftText         | copy on the left                                                         | String  | -       |   
| fixed         | Is it fixed                                                         | Boolean  | false       |   
| safeAreaInsetTop         | Whether it is suitable for the safe area                                                         | Boolean  | false       |   
| border         | whether to show the bottom border                                      | Boolean  | false    | 
| placeholder         | When fixed to the top, whether to generate a placeholder element of equal height at the label position           | Boolean  | false    |
| zIndex         | Navigation Bar Hierarchy           | Number„ÄÅString  | 10    |
| style         | container style           | React.CSSProperties  | {}    |
| className         | container class name           | String  | ""    |                                          

### Event

| Event  | Description     | callback parameter    |
|-------|----------|-------------|
| onClickTitle | click title event | event:Event |
| onClickRight | Click on the event on the right | event:Event |
| onClickBack | click back event | event:Event |
| onClickIcon | Click the icon event on the right side of the title | event:Event |