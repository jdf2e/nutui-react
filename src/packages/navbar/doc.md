# Navbar 头部导航

## 介绍

提供导航功能。

## 安装

```tsx
import { NavBar } from '@nutui/nutui-react';
```

## 代码演示

### 基础用法

:::demo

```tsx
import  React from "react";
import { NavBar, Toast } from '@nutui/nutui-react';
import { Left, Share, Close } from '@nutui/icons-react'

const App = () => {
  return ( 
    <NavBar
        back={
        <>
            <Left name="left" color="#979797" />
            返回
        </>
        }
        left={<Close width={12} />}
        right={
        <span onClick={(e) =>  Toast.show('icon')}>
            <Share />
        </span>
        }
        onBackClick={(e) =>  Toast.show("返回")}
    >
        <span onClick={(e) =>  Toast.show("标题")}>
        订单详情
        </span>
    </NavBar>
  );
};  
export default App;

```

:::

:::demo

```tsx
import  React from "react";
import { NavBar, Toast } from '@nutui/nutui-react';
import { Left } from '@nutui/icons-react'

const App = () => {
  return ( 
    <NavBar
        right={
        <span onClick={(e) =>  Toast.show('清空')}>
            清空
        </span>
        }
        back={<Left name="left" color="#979797" />}
        onBackClick={(e) =>  Toast.show("返回")}
    >
        <span onClick={(e) =>  Toast.show("标题")}>
        浏览记录
        </span>
    </NavBar>
  );
};  
export default App;

```

:::

:::demo

```tsx
import  React from "react";
import { NavBar, Toast } from '@nutui/nutui-react';
import { Cart2, Left, MoreX} from '@nutui/icons-react'

const App = () => {
  return ( 
    <NavBar
        back={<Left name="left" color="#979797" />}
        right={
        <>
            <span style={{ marginRight: '5px' }} onClick={(e) =>  Toast.show('编辑')}>
            编辑
            </span>
            <MoreX onClick={(e) =>  Toast.show('icon')} />
        </>
        }
        onBackClick={(e) =>  Toast.show("返回")}
    >
        <span onClick={(e) =>  Toast.show("标题")}>
        购物车
        </span>
        <i style={{ marginLeft: '5px' }} onClick={(e) =>  Toast.show('icon')}>
            <Cart2 />
        </i>
    </NavBar>
  );
};  
export default App;

```

:::

:::demo

```tsx
import  React, { useState } from "react";
import { NavBar, Tabs, TabPane, Toast } from '@nutui/nutui-react';
import { Left,MoreX } from '@nutui/icons-react'

const App = () => {
  const [tab1value, setTab1value] = useState('0')
  return (   
      <NavBar
         back={<Left name="left" color="#979797" />}
          right={
            <>
              <span style={{ marginRight: '5px' }}  onClick={(e) =>  Toast.show("编辑")}>
                编辑
              </span>
              <MoreX onClick={(e) =>  Toast.show('icon')} />
            </>
          }
          onBackClick={(e) =>  Toast.show("返回")}
        >
            <Tabs value={tab1value} onChange={({ paneKey }) => { setTab1value(paneKey) }}>
              <TabPane title="Tab 1"> Tab 1 </TabPane>
              <TabPane title="Tab 2"> Tab 2 </TabPane>
              <TabPane title="Tab 3"> Tab 3 </TabPane>
            </Tabs>
      </NavBar>
  );
};  
export default App;

```

:::

## Navbar

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| right | 右侧内容 | `ReactNode` | `-` |
| left | 左侧内容，渲染在返回区域的右侧 | `ReactNode` | `-` |
| back | 返回区域的文字 | `ReactNode` | `-` |
| fixed | 是否固定 | `boolean` | `false` |
| safeArea | 是否适配安全区 | `boolean` | `false` |
| placeholder | 固定在顶部时，是否在标签位置生成一个等高的占位元素 | `boolean` | `false` |
| zIndex | 导航栏层级 | `number` \| `string` | `10` |
| onBackClick | 点击返回区域后的回调 | `onBackClick:(event: Event)=>void` | `false` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 说明 | 默认值 |
| --- | --- | --- |
| \--nutui-navbar-width | 头部导航的宽度 | `100%` |
| \--nutui-navbar-height | 头部导航的高度 | `44px` |
| \--nutui-navbar-margin-bottom | 头部导航的下边距 | `20px` |
| \--nutui-navbar-padding | 头部导航的内边距 | `13px 16px` |
| \--nutui-navbar-background | 头部导航的背景颜色 | `$white` |
| \--nutui-navbar-box-shadow | 头部导航的阴影 | `0px 1px 7px 0px rgba(237, 238, 241, 1)` |
| \--nutui-navbar-color | 头部导航的字体颜色 | `$gray2` |
| \--nutui-navbar-font-size | 头部导航的字体大小 | `$font-size-2` |
| \--nutui-navbar-title-font-size | 头部导航标题的字体大小 | `$font-size-2` |
| \--nutui-navbar-title-font-weight | 头部导航标题的字体粗细 | `0` |
| \--nutui-navbar-title-font-color | 头部导航标题的字体颜色 | `$gray1` |