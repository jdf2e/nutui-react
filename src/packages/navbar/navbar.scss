.nut-navbar {
  width: $navbar-width;
  position: relative;
  display: flex;
  align-items: center;
  height: $navbar-height;
  box-sizing: border-box;
  background: $navbar-background;
  box-shadow: $navbar-box-shadow;
  font-size: $navbar-font-size;
  color: $navbar-color;
  margin-bottom: $navbar-margin-bottom;
  overflow: hidden;

  &-fixed {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
  }

  &-placeholder {
    display: inline-block;
    width: 100%;
  }

  &-safe-area-inset-top {
    padding-top: constant(safe-area-inset-top);
    padding-top: env(safe-area-inset-top);
  }

  &-title {
    min-width: 53%;
    margin: 0 auto;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: $navbar-title-font-size;
    font-weight: $navbar-title-font-weight;
    color: $navbar-title-font-color;

    .nut-tabs-titles {
      background: transparent;

      &-item-text {
        color: $color-title;
      }
    }

    .nut-tabpane {
      display: none;
    }
  }

  &-title ::-webkit-scrollbar {
    display: none;
  }

  &-left,
  &-right {
    position: absolute;
    top: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    cursor: pointer;
    padding: 0 14px;

    .nut-icon,
    .nutui-iconfont {
      width: 20px;
      height: 20px;
      font-size: 20px;
    }
  }

  &-left {
    left: 0;

    &-back {
      display: flex;
      align-items: center;
      justify-content: center;
      margin-right: 10px;
    }
  }

  &-right {
    right: 0;

    .nut-icon {
      margin-left: 12px;

      &:first-child {
        margin-left: 0;
      }
    }
  }

  &-title-align-left {
    padding-inline-start: 14px;

    .nut-navbar-left {
      position: static;
      padding: 0;

      .nut-icon {
        margin-left: 12px;

        &:first-child {
          margin-left: 0;
        }
      }
    }
    .nut-navbar-right {
      position: static;
      display: inline-flex;
      white-space: nowrap;
      padding-inline-start: 0;
    }
    .nut-navbar-left-back {
      margin-right: 0;
    }

    .nut-navbar-title {
      min-width: 0;
      margin: 0;
      flex: 1;
      justify-content: flex-start;
      padding: 0 8px;
    }
  }
}

[dir='rtl'] .nut-navbar,
.nut-rtl .nut-navbar {
  &-fixed {
    left: auto;
    right: 0;
  }

  &-left {
    left: auto;
    right: 0;
    .nut-icon-ArrowLeft {
      transform: rotateY(180deg);
    }
  }

  &-right {
    right: auto;
    left: 0;
  }
}
