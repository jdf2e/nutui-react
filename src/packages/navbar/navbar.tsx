import React, { FunctionComponent, useEffect, useRef, useState } from 'react'
import classNames from 'classnames'

import { BasicComponent, ComponentDefaults } from '@/utils/typings'
import { getRect } from '@/utils/use-client-rect'

export interface NavBarProps extends BasicComponent {
  left: React.ReactNode
  back: React.ReactNode
  right: React.ReactNode
  titleAlign: 'center' | 'left'
  fixed: boolean
  safeAreaInsetTop: boolean
  placeholder: boolean
  zIndex: number | string
  onBackClick: (e: React.MouseEvent<HTMLElement>) => void
  children?: React.ReactNode
}

const defaultProps = {
  ...ComponentDefaults,
  left: '',
  right: '',
  titleAlign: 'center',
  back: '',
  fixed: false,
  safeAreaInsetTop: false,
  placeholder: false,
  zIndex: 10,
} as NavBarProps
export const NavBar: FunctionComponent<Partial<NavBarProps>> = (props) => {
  const {
    right,
    left,
    titleAlign,
    className,
    style,
    back,
    fixed,
    safeAreaInsetTop,
    placeholder,
    zIndex,
    onBackClick,
  } = {
    ...defaultProps,
    ...props,
  }

  const classPrefix = 'nut-navbar'

  const children = Array.isArray(props.children)
    ? props.children
    : [props.children]

  const styles = () => {
    return {
      ...style,
      zIndex,
    }
  }

  const leftRef = useRef<HTMLDivElement>(null)
  const rightRef = useRef<HTMLDivElement>(null)
  const wrapperRef = useRef<HTMLDivElement>(null)
  const [contentWidth, setContentWidth] = useState('50%')

  const getNodeWidth = (node: Element | null) => {
    if (node) {
      const ele = getRect(node)
      return ele.width
    }
    return 0
  }

  useEffect(() => {
    if (titleAlign === 'left') {
      return
    }
    if (!(back || left || right)) {
      setContentWidth('100%')
      return
    }
    const leftRectWidth = getNodeWidth(leftRef?.current)
    const rightRectWidth = getNodeWidth(rightRef?.current)
    const wrapperWidth = getNodeWidth(wrapperRef?.current)

    let centerWidth = wrapperWidth / 2
    if (leftRectWidth && rightRectWidth) {
      centerWidth =
        wrapperWidth -
        (leftRectWidth > rightRectWidth
          ? leftRectWidth * 2
          : rightRectWidth * 2)
    } else {
      centerWidth = wrapperWidth - leftRectWidth * 2 - rightRectWidth * 2
    }

    setContentWidth(centerWidth.toFixed(2))
  }, [left, right, back, titleAlign])

  const renderLeft = () => {
    return back || left ? (
      <div className={`${classPrefix}-left`} ref={leftRef}>
        {back && (
          <div
            className={`${classPrefix}-left-back`}
            onClick={(e) => onBackClick(e)}
          >
            {back}
          </div>
        )}
        {left}
      </div>
    ) : null
  }

  const renderContent = () => {
    let titleStyle = {}
    if (titleAlign === 'center') {
      const contentRealWidth = `${contentWidth}${
        /%$/i.test(contentWidth) ? '' : 'px'
      }`
      titleStyle = {
        minWidth: contentRealWidth,
        width: contentRealWidth,
      }
    }

    return (
      <div className={`${classPrefix}-title`} style={titleStyle}>
        {children}
      </div>
    )
  }

  const renderRight = () => {
    return (
      <div className={`${classPrefix}-right`} ref={rightRef}>
        {right}
      </div>
    )
  }

  const renderWrapper = () => {
    return (
      <div className={cls} style={styles()} ref={wrapperRef}>
        {renderLeft()}
        {renderContent()}
        {renderRight()}
      </div>
    )
  }

  const classes = classNames({
    [`${classPrefix}-fixed`]: fixed,
    [`${classPrefix}-safe-area-inset-top`]: safeAreaInsetTop,
    [`${classPrefix}-title-align-${titleAlign}`]: true,
  })

  const cls = classNames(classPrefix, classes, className)

  return (
    <>
      {fixed && placeholder ? (
        <div className={`${classPrefix}-placeholder`}>{renderWrapper()}</div>
      ) : (
        renderWrapper()
      )}
    </>
  )
}

NavBar.displayName = 'NutNavBar'
