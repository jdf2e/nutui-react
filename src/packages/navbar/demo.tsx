import React, { useState } from 'react'
import { Share, MoreX, Cart2, Left, Close } from '@nutui/icons-react'
import { NavBar } from './navbar'
import { Tabs } from '../tabs/tabs'
import { TabPane } from '../tabpane/tabpane'
import { useTranslate } from '../../sites/assets/locale'
import Toast from '../toast/toast'

interface T {
  ce5c5446: string
  c38a08ef: string
  b840c88f: string
  a74a1fd4: string
  '8dab2f66': string
  cfbdc781: string
  c3a3a1d2: string
  e51e4582: string
}

const NavBarDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      ce5c5446: '基础用法',
      c38a08ef: '订单详情',
      b840c88f: '标题',
      a74a1fd4: '返回',
      '8dab2f66': '编辑',
      cfbdc781: '清空',
      c3a3a1d2: '购物车',
      e51e4582: '浏览记录',
    },
    'zh-TW': {
      ce5c5446: '基礎用法',
      c38a08ef: '訂單詳情',
      b840c88f: '標題',
      a74a1fd4: '返回',
      '8dab2f66': '編輯',
      cfbdc781: '清空',
      c3a3a1d2: '購物車',
      e51e4582: '瀏覽記錄',
    },
    'en-US': {
      ce5c5446: 'Basic usage',
      c38a08ef: 'order details',
      b840c88f: 'title',
      a74a1fd4: 'return',
      '8dab2f66': 'edit',
      cfbdc781: 'empty',
      c3a3a1d2: 'shopping cart',
      e51e4582: 'Browsing history',
    },
  })
  const [tab1value, setTab1value] = useState<string | number>('0')
  return (
    <>
      <div className="demo">
        <h2>{translated.ce5c5446}</h2>
        <NavBar
          back={
            <>
              <Left name="left" color="#979797" />
              {translated.a74a1fd4}
            </>
          }
          left={<Close width={12} />}
          right={
            <span onClick={(e) => Toast.show('icon')}>
              <Share />
            </span>
          }
          onBackClick={(e) => Toast.show(translated.a74a1fd4)}
        >
          <span onClick={(e) => Toast.show(translated.b840c88f)}>
            {translated.c38a08ef}
          </span>
        </NavBar>

        <NavBar
          right={
            <span onClick={(e) => Toast.show(translated.cfbdc781)}>
              {translated.cfbdc781}
            </span>
          }
          back={<Left name="left" color="#979797" />}
          onBackClick={(e) => Toast.show(translated.a74a1fd4)}
        >
          <span onClick={(e) => Toast.show(translated.b840c88f)}>
            {translated.e51e4582}
          </span>
        </NavBar>
        <NavBar
          back={<Left name="left" color="#979797" />}
          right={
            <>
              <span onClick={(e) => Toast.show(translated['8dab2f66'])}>
                {translated['8dab2f66']}
              </span>
              <MoreX onClick={(e) => Toast.show('icon')} />
            </>
          }
          onBackClick={(e) => Toast.show(translated.a74a1fd4)}
        >
          <span onClick={(e) => Toast.show(translated.b840c88f)}>
            {translated.c3a3a1d2}
          </span>
          <i style={{ marginLeft: '5px' }} onClick={(e) => Toast.show('icon')}>
            <Cart2 />
          </i>
        </NavBar>

        <NavBar
          back={<Left name="left" color="#979797" />}
          right={
            <>
              <span onClick={(e) => Toast.show(translated['8dab2f66'])}>
                {translated['8dab2f66']}
              </span>
              <MoreX onClick={(e) => Toast.show('icon')} />
            </>
          }
          onBackClick={(e) => Toast.show(translated.a74a1fd4)}
        >
          <div slot="content">
            <Tabs
              value={tab1value}
              onChange={(paneKey) => {
                setTab1value(paneKey)
              }}
            >
              <TabPane title="Tab 1"> Tab 1 </TabPane>
              <TabPane title="Tab 2"> Tab 2 </TabPane>
              <TabPane title="Tab 3"> Tab 3 </TabPane>
            </Tabs>
          </div>
          <i slot="right">
            <MoreX />
          </i>
        </NavBar>
      </div>
    </>
  )
}

export default NavBarDemo
