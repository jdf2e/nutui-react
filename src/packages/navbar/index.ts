import { NavBar } from './navbar'

export type { NavBarProps } from './navbar'
export default NavBar
