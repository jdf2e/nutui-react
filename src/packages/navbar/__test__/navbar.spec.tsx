import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import { NavBar } from '../navbar'

test('should render left slot correctly', () => {
  const { container } = render(<NavBar left={<span>left</span>} />)
  expect(container.querySelectorAll('.nut-navbar-left')[0].innerHTML).toContain(
    '<span>left</span>'
  )
})

test('should render right slot correctly', () => {
  const { container } = render(<NavBar right={<span>right</span>} />)
  expect(
    container.querySelectorAll('.nut-navbar-right')[0].innerHTML
  ).toContain('<span>right</span>')
})

test('should render title slot correctly', () => {
  const { container } = render(
    <NavBar>
      <span>content</span>
    </NavBar>
  )
  expect(
    container.querySelectorAll('.nut-navbar-title')[0].innerHTML
  ).toContain('<span>content</span>')
})

test('should left-text', () => {
  const { container } = render(<NavBar left="back">订单详情</NavBar>)
  expect(container.querySelectorAll('.nut-navbar-left')[0].innerHTML).toBe(
    'back'
  )
})

test('should description', () => {
  const { container } = render(<NavBar right="description">订单详情</NavBar>)
  expect(container.querySelectorAll('.nut-navbar-right')[0].innerHTML).toBe(
    'description'
  )
})

test('should render placeholder element when using placeholder prop', () => {
  const { container } = render(
    <NavBar fixed placeholder>
      订单详情
    </NavBar>
  )
  expect(
    container.querySelectorAll('.nut-navbar-placeholder')[0].innerHTML
  ).toMatchSnapshot()
})

test('should emit click-back event when clicking back text', () => {
  const onBackClick = vi.fn()
  const { container } = render(
    <NavBar back="返回" onBackClick={onBackClick}>
      订单详情
    </NavBar>
  )

  fireEvent.click(container.querySelectorAll('.nut-navbar-left-back')[0])
  expect(onBackClick).toBeCalled()
})

test('should change z-index when using z-index prop', () => {
  const { container } = render(<NavBar zIndex="100">订单详情</NavBar>)
  expect((container.firstChild as HTMLDivElement).style.zIndex).toBe('100')
})
