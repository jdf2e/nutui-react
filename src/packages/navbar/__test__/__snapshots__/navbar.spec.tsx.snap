// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render placeholder element when using placeholder prop 1`] = `"<div class=\\"nut-navbar nut-navbar--fixed\\" style=\\"z-index: 10;\\"><div class=\\"nut-navbar__left\\"><i class=\\" nutui-iconfont nut-icon nutui-icon nut-icon-left \\" style=\\"color: rgb(151, 151, 151);\\"></i></div><div class=\\"nut-navbar__title\\"><div class=\\"title\\">订单详情</div></div><div class=\\"nut-navbar__right\\"></div></div>"`;
